### PR TITLE
Animation refactors tier 3

### PR DESCRIPTION
**Pull request by Sezzary**
Part 3 of animation refactors.
- Replaces almost all `g_level.Anims` access with `GetAnimData()`, readying the codebase for the complete removal of the `g_Level.Anims` vector in tier 4.
- Makes all access to animation data stored in the level strictly read-only. Previously, some objects directly and inappropriately modified frame data, which blocked this refactor from happening until now.

Things to test:
- Flare, torch.
- Rope.
- Interactable objects.
- Plinth pickup.
- Others at your discretion if anything looks suspicious in the modified files.

### CODE REVIEW

File: TombEngine/Objects/TR4/Object/WraithTrap.cpp [hunk 1]
```diff
@@ -22,8 +22,8 @@ namespace TEN::Entities::TR4
 	{
 		auto& item = g_Level.Items[itemNumber];
 
-		item.Animation.AnimNumber = Objects[item.ObjectNumber].animIndex;
-		item.Animation.FrameNumber = g_Level.Anims[item.Animation.AnimNumber].frameBase;
+		item.Animation.AnimNumber = Objects[item.ObjectNumber].animIndex + 3;
```
Comment (Lwmte) [inline TombEngine/Objects/TR4/Object/WraithTrap.cpp:25]: Why a magic number 3 was added here?

Comment (Sezzary) [reply TombEngine/Objects/TR4/Object/WraithTrap.cpp:25]: May have been a stray Crtl+V. Should use `SetAnimation()` here.

Comment (Lwmte) [reply TombEngine/Objects/TR4/Object/WraithTrap.cpp:25]: Yes please :)

**Review by Lwmte — APPROVED** [review]

**Comment by Stranger1992** [comment]
Well that was a boring test. I didn't find anything that was wrong or incorrect so approved by me!

**Review by Stranger1992 — APPROVED** [review]

**Pull request merged**